File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minelement = prices[0];
        int mxsum = 0;
        for(int i = 0 ; i < prices.size();i++){
            minelement = min(minelement,prices[i]);
            mxsum = max(mxsum,prices[i]-minelement);
        }
        return mxsum;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:
        minid = nums[0]
        maxprofit = 0
        
        for i in range(1,len(nums)):
            maxprofit = max(maxprofit,nums[i]-minid)
            minid = min(minid,nums[i])

        return maxprofit
            